distribution/pull_v2.go

```diff
@@ -665,14 +665,20 @@ func (p *v2Puller) pullSchema2Layers(ctx context.Context, target distribution.De
 		}
 	}
 
+	// Assume that the operating system is the host OS if blank, and validate it
+	// to ensure we don't cause a panic by an invalid index into the layerstores.
+	if layerStoreOS != "" && !system.IsOSSupported(layerStoreOS) {
+		return "", system.ErrNotSupportedOperatingSystem
+	}
+
 	if p.config.DownloadManager != nil {
```

Comment:
"technically" the check should be within this branch (to closer match the previous behavior)

I'm not sure in which case `p.config.DownloadManager` _would_ be allowed to be `nil` though; looking when this check was added, I see that was part of the commit that introduced the Interface (https://github.com/moby/moby/commit/3c7676a057a4c0103895f793e407dc6736df139a / https://github.com/moby/moby/pull/29339), and (as far as I can tell) is mostly a "sanity check" to prevent a panic (perhaps it should just return an error if we end up in this situation?)